core/arch/x86/encode.c

```diff
@@ -1053,6 +1041,12 @@ reg_size_ok(decode_info_t *di /*prefixes field is IN/OUT; x86_mode is IN*/, reg_
         }
         return true;
     }
+    /* XXX i#1312: we haven't really resolved the register's size. The mask register
+     * size is dependent on the machine's support for certain AVX-512 derivatives. We may
+     * want to look at XCR0 bits and resolve the size in a new function similar to
```

Comment:
Isn't it whether AVX512BW is present, which is cpuid bits, not XCR0 bits, for whether mask registers are 16 or 64 bits?

Reply:
That's true. In comment, s/XCR0/cpuid/.

Reply:
That said, a decoder/encoder should not act based on the underlying processor: it should be completely controlled by parameters set by the user.  It should be able to decode/encode across platforms and for modes not present on the current machine.